src/Strategy/IfElse.php

```diff
@@ -0,0 +1,55 @@
+<?php
+namespace ScriptFUSION\Mapper\Strategy;
+
+use ScriptFUSION\Mapper\Mapping;
+
+/**
+ * Delegates to one expression or another depending on whether the specified condition loosely evaluates to true.
+ */
+class IfElse extends Delegate
+{
+    /** @var Strategy|Mapping|array|mixed */
+    private $if;
+
+    /** @var Strategy|Mapping|array|mixed */
+    private $else;
+
+    /**
+     * Initializes this instance with the specified condition, the specified
+     * strategy or mapping to be resolved when condition is non-null and,
+     * optionally, the specified strategy or mapping to be resolved when
+     * condition is null.
+     *
+     * @param Strategy|Mapping|array|mixed $condition Condition.
```

Comment:
It seems this type union disagrees with the documentation. However, as discussed, let's use `callable` here.